x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { nullableType } from './lib/nullable';
+import { portSchema } from './lib/schemas';
+import { isOk, promiseResult, Result } from './lib/result_type';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+
+// config definition
+enum WebhookMethods {
+  POST = 'post',
+  PUT = 'put',
+}
+enum WebhookSchemes {
+  HTTP = 'http',
+  HTTPS = 'https',
+}
+
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const HeadersSchema = schema.recordOf(schema.string(), schema.string());
+
+export const CompositeUrlSchema = schema.object({
+  host: schema.string(),
+  port: portSchema(),
+  path: nullableType(schema.string()),
+  scheme: schema.oneOf(
+    [schema.literal(WebhookSchemes.HTTP), schema.literal(WebhookSchemes.HTTPS)],
+    {
+      defaultValue: WebhookSchemes.HTTP,
+    }
+  ),
+});
+
+const ConfigSchema = schema.object({
+  url: schema.oneOf([schema.string(), CompositeUrlSchema]),
+  method: schema.oneOf([schema.literal(WebhookMethods.POST), schema.literal(WebhookMethods.PUT)], {
+    defaultValue: WebhookMethods.POST,
+  }),
+  headers: nullableType(HeadersSchema),
+});
+
+// secrets definition
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const SecretsSchema = schema.object({
+  user: schema.string(),
+  password: schema.string(),
+});
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  body: schema.maybe(schema.string()),
+});
+
+// action type definition
+export const actionType: ActionType = {
+  id: '.webhook',
+  name: 'webhook',
+  validate: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+    params: ParamsSchema,
+  },
+  executor,
+};
+
+function asComposedUrl(url: string | TypeOf<typeof CompositeUrlSchema>): string {
+  if (typeof url === 'string') {
+    return url;
+  }
+  return `${url.scheme}://${url.host}${url.port ? `:${url.port}` : ''}${url.path || ''}`;
+}
+
+function describeWebhookErrorSource(error: AxiosError): string {
+  if (error.response) {
+    // The request was made and the server responded with a status code
+    // that falls out of the range of 2xx
+    return `remote webhook failure`;
```

Comment:
Would be good to include the http status here.

Also, I guess these need to get translated?

Reply:
Yeah, I hadn't quite twigged the translation mechanism when i wrote that.
I've now unified the handlers for these cases and it's now possible to translate the "unreachable remote" which is the *one* case I don't actually have a status code/text for because Axios just omits the response object and request is an `any`, so who knows what they put in there.